src/gui/clips/PatternClipView.cpp

```diff
@@ -115,14 +115,18 @@ void PatternClipView::paintEvent(QPaintEvent*)
 	
 	// bar lines
 	const int lineSize = 3;
+	int pixelsPerPattern = Engine::patternStore()->lengthOfPattern(m_patternClip->patternIndex()) * pixelsPerBar();
+	int offset = static_cast<int>(m_patternClip->startTimeOffset() * (pixelsPerBar() / TimePos::ticksPerBar()))
+			% pixelsPerPattern;
+	if (offset <= 1) {
+		offset += pixelsPerPattern;
+	}
+
 	p.setPen( c.darker( 200 ) );
 
-	bar_t t = Engine::patternStore()->lengthOfPattern(m_patternClip->patternIndex());
-	if (m_patternClip->length() > TimePos::ticksPerBar() && t > 0)
+	if (pixelsPerPattern > 0 && offset < width() - 2)
```

Comment:
I think the `offset < width() - 2` condition here is superfluous: the loop won't run anyway if that's not the case. But there's nothing wrong about it, so I'm happy for it to be left as-is.

Reply:
No problem, I will delete it. I thought it might be good to not initiate an unnecessary loop but it shouldn't make a big difference in performance here. So it's better to keep it clean for others. 👍